### PR TITLE
DT-2500 Make sure same data is returned for bookings and offenders endpoints

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/test/resources/uk/gov/justice/hmpps/prison/api/resource/impl/offender_extra_info.json [hunk 1]
```diff
@@ -154,6 +154,7 @@
   "status": "ACTIVE IN",
   "legalStatus": "IMMIGRATION_DETAINEE",
   "imprisonmentStatus": "DEPORT",
+  "receptionDate": "2021-09-06",
```
Comment (Mjwillis) [inline src/test/resources/uk/gov/justice/hmpps/prison/api/resource/impl/offender_extra_info.json:157]: This file is used by multiple test for pulling data back in different endpoints.  Data set should be the same for all.

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java [hunk 2]
```diff
@@ -438,6 +438,14 @@ public void getFullOffenderInformation_byOffenderNo() {
         assertThatJsonFileAndStatus(response, 200, "offender_extra_info.json");
     }
 
+    @Test
+    public void getFullOffenderInformation_byOffenderNoAlt() {
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java:442]: what does `noAlt` refer to in `getFullOffenderInformation_byOffenderNoAlt()` ?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt [hunk 3]
```diff
@@ -3,30 +3,31 @@ package uk.gov.justice.hmpps.prison.repository.sql
...
+        O.OFFENDER_ID_DISPLAY OFFENDER_NO,
         O.FIRST_NAME,
         CONCAT(O.middle_name, CASE WHEN middle_name_2 IS NOT NULL THEN concat(' ', O.middle_name_2) ELSE '' END) MIDDLE_NAME,
         O.LAST_NAME,
-        B.AGY_LOC_ID,
-        B.LIVING_UNIT_ID,
-        B.ACTIVE_FLAG,
+        B.AGY_LOC_ID AGENCY_ID,
+        B.LIVING_UNIT_ID ASSIGNED_LIVING_UNIT_ID,
+        CASE WHEN B.ACTIVE_FLAG = 'Y' THEN 1 ELSE 0 END ACTIVE_FLAG,
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:16]: could this potentially break existing clients? changing type from char to boolean? 

Or is this handle by some other layer?

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt [hunk 4]
```diff
@@ -3,30 +3,31 @@ package uk.gov.justice.hmpps.prison.repository.sql
...
-        AND OI.ORIENTATION_TYPE = 'FRONT') AS FACE_IMAGE_ID,
-        O.BIRTH_DATE,
+                AND OI.IMAGE_VIEW_TYPE = 'FACE'
+        AND OI.ORIENTATION_TYPE = 'FRONT') AS FACIAL_IMAGE_ID,
+        O.BIRTH_DATE DATE_OF_BIRTH,
         O.BIRTH_PLACE,
         O.BIRTH_COUNTRY_CODE,
-        B.ASSIGNED_STAFF_ID AS ASSIGNED_OFFICER_ID
+        B.ASSIGNED_STAFF_ID AS ASSIGNED_OFFICER_ID,
+        B.BOOKING_BEGIN_DATE AS RECEPTION_DATE
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:30]: I am confused, so was reception date always missing? I though it was present sometimes?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt (hunk 3 above)
Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:16]: guess it was broken previously eitherway

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:16]: It used a different mapper

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java (hunk 2 above)
Comment (Mjwillis) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.java:442]: Alt was alternative!  Sorry horrible wording

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt (hunk 4 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:30]: Yes present in other endpoint

**Pull request merged**